tests/unittests/RecommendationSystemTest.cpp

```diff
@@ -733,13 +749,29 @@ class RecommendationSystemTest : public BackendTest {
       auto *saveConcat = F_->createSave("after_concat_data", CN);
       concatPH = saveConcat->getPlaceholder();
     }
+    std::vector<std::unique_ptr<DeviceConfig>> configs;
+    auto deviceConfig = llvm::make_unique<DeviceConfig>(getBackendName());
+    deviceConfig->setDeviceMemory(64e+9);
+    configs.push_back(std::move(deviceConfig));
+    std::unique_ptr<HostManager> hostManager(
+        new HostManager(std::move(configs)));
 
     CompilationContext cctx;
     cctx.precisionConfig = precConfig_;
-    EE_.compile(cctx);
+    hostManager->addNetwork(std::move(mod), cctx);
 
     // Run graph
-    EE_.run(context_);
+    // Create a second context and run it too to test concurrent requests.
+    ExecutionContext context2{};
+    std::promise<void> runNetwork, runNetwork2;
+    auto ready = runNetwork.get_future();
+    auto ready2 = runNetwork2.get_future();
+    dispatchInference(hostManager.get(), runNetwork, context_);
```

Comment:
Does it mean we now unconditionally always run two concurrent requests? May be it should be done conditionally?

Reply:
Refactored it into a command-line arg concurrent-requests